### ISSUE TITLE
configmap change fail when changing global tags

### ISSUE THREAD

**Issue by ArchiFleKs** [terraform-aws-modules/terraform-aws-eks#1449]
## Description

When changing cluster tag and running apply ( tested in TF 0.14.10 and 1.0.0) even when forcing a refresh before the configmmap map refresh fail with the classic:

```
╷                                                                                                                                                                                                                                                                                                                                                                                                                                      
│ Error: Get "http://localhost/api/v1/namespaces/kube-system/configmaps/aws-auth": dial tcp 127.0.0.1:80: connect: connection refused                                                                                                                                                                                                                                                                                                  
│                                                                                                                                                                                                                                                                                                                                                                                                                                      
│   with kubernetes_config_map.aws_auth[0],                                                                                                                                                                                                                                                                                                                                                                                            
│   on aws_auth.tf line 63, in resource "kubernetes_config_map" "aws_auth":                                                                                                                                                                                                                                                                                                                                                            
│   63: resource "kubernetes_config_map" "aws_auth" {                                                                                                                                                              
│                                                                                            
╵                                                
```

## Versions
<!-- REQUIRED -->
 - Terraform: 1.0.0
<!-- Execute terraform -version -->
 - Provider(s):
 ```
Terraform v1.0.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.43.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.3.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/terraform-aws-modules/http v2.4.1
```
<!-- Execute: terraform providers -version -->
 - Module:

## Reproduction
<!-- REQUIRED -->
<!-- Are you using workspaces? -->
<!-- Have you cleared the local cache (see Notice section above)? -->
<!-- List steps in order that led up to the issue you encountered -->

### Code Snippet to Reproduce


Change the  `tags` values on the module

## Expected behavior
<!-- A clear and concise description of what you expected to happen -->

Configmap is refreshed

## Actual behavior
<!-- A clear and concise description of what actually happened -->

```
╷                                                                                                                                                                                                                                                                                                                                                                                                                                      
│ Error: Get "http://localhost/api/v1/namespaces/kube-system/configmaps/aws-auth": dial tcp 127.0.0.1:80: connect: connection refused                                                                                                                                                                                                                                                                                                  
│                                                                                                                                                                                                                                                                                                                                                                                                                                      
│   with kubernetes_config_map.aws_auth[0],                                                                                                                                                                                                                                                                                                                                                                                            
│   on aws_auth.tf line 63, in resource "kubernetes_config_map" "aws_auth":                                                                                                                                                                                                                                                                                                                                                            
│   63: resource "kubernetes_config_map" "aws_auth" {                                                                                                                                                                                                                                                                                                                                                                                  
│                                                                                                                                                                                                                                                                                                                                                                                                                                      
╵                                                                                                                                                                                                                                                                                                                                                                                                                                      
ERRO[0012] Hit multiple errors:              
```

## Additional context

I'm using Terragrunt, here is my provider block:

```
    provider "aws" {
      region = "${local.aws_region}"
    }
    provider "kubernetes" {
      host                   = data.aws_eks_cluster.cluster.endpoint
      cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
      token                  = data.aws_eks_cluster_auth.cluster.token
    }
    data "aws_eks_cluster" "cluster" {
      name = aws_eks_cluster.this[0].id
    }
    data "aws_eks_cluster_auth" "cluster" {
      name = aws_eks_cluster.this[0].id
    }
```

**Comment by ArchiFleKs** [comment]
> Just commenting that I have run into this issue multiple times in the past when trying to alter tags. I don't use this module, so I suspect the issue may be related to the kubernetes provider itself. I'm using the latest v2.x.x provider version.
> 
> Currently the only way we can change the tags is to destroy the cluster and re-create it with the updated tags.
> 
> If anyone has a better workaround, please let me know!

You can set manage AWS auth to false in module. Then remove the config map from the state. Then apply your changes. Then switch the flag to true. Finally reimport config map in state

**Comment by igorinsky** [comment]
We've spotted the same issue in attempts to change EKS cluster version - terragrunt failed to authorize while refreshing for kubernetes_config_map.aws_auth[0].

But it doesn't fail when you do other operations (it refreshes the state without errors) or even modify map_roles, which is all about this resource.

Our workaround was to add config_path to kubernetes provider identical to kubeconfig_output_path, e.g.
```
provider "kubernetes" {
  config_path = "/tmp/kubeconfig"
}
```
and move back to 
```
provider "kubernetes" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}
```
once changes were applied.

P.S. Still seems like a bug. FYI module version we used was 17.0.3
